examples/end_to_end_distance.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2015, Ecole Polytechnique Federale de Lausanne, Blue Brain Project
+# All rights reserved.
+#
+# This file is part of NeuroM <https://github.com/BlueBrain/NeuroM>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     1. Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#     2. Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in the
+#        documentation and/or other materials provided with the distribution.
+#     3. Neither the name of the copyright holder nor the names of
+#        its contributors may be used to endorse or promote products
+#        derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''Calculate and plot end-to-end distance of neurites.'''
+
+from neurom.analysis import morphmath
+from neurom import ezy
+from neurom.core import tree
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def furthest_leaf(neurite):
+    '''Find and return the neurite point that is the furthest away from the
+    neurite trunk.'''
+    trunk = neurite.value[0:3]
+    furthest = trunk
+    max_distance = 0
+    for leaf in tree.val_iter(tree.ileaf(neurite)):
+        new_distance = morphmath.point_dist(leaf[0:3], trunk)
+        if new_distance > max_distance:
+            max_distance = new_distance
+            furthest = leaf[0:3]
+    return furthest
```

Comment:
I think this can be simplified to

trunk = neurite.value
return min((morphmath.point_dist(l, trunk) for l in tree.val_iter(tree.ileaf(n0)))